plugins/telemetry-otel/src/main/java/org/opensearch/telemetry/tracing/sampler/RequestSampler.java

```diff
@@ -18,21 +22,24 @@
 import io.opentelemetry.sdk.trace.samplers.Sampler;
 import io.opentelemetry.sdk.trace.samplers.SamplingResult;
 
+import static org.opensearch.telemetry.OTelTelemetrySettings.OTEL_TRACER_SPAN_SAMPLER_CLASS_SETTINGS;
+import static org.opensearch.telemetry.tracing.AttributeNames.TRACE;
+
 /**
- * HeadBased sampler
+ * RequestSampler based on HeadBased sampler
  */
 public class RequestSampler implements Sampler {
-    private final Sampler defaultSampler;
-
-    // TODO: Pick value of TRACE from PR #9415.
-    private static final String TRACE = "trace";
+    private final HashMap<String, Sampler> samplers;
+    private final List<String> samplerList;
 
     /**
-     * Creates Head based sampler
-     * @param defaultSampler defaultSampler
+     * Creates request sampler which applies based on all applicable sampler
+     * @param telemetrySettings TelemetrySettings
+     * @param setting Settings
      */
-    public RequestSampler(Sampler defaultSampler) {
-        this.defaultSampler = defaultSampler;
+    public RequestSampler(TelemetrySettings telemetrySettings, Settings setting) {
```

Comment:
It should have a parameter of List<Sampler> otelSamplers instead of creating through the factory inside the constructor.